br/pkg/lightning/backend/local/localhelper.go

```diff
@@ -430,10 +439,6 @@ func (local *Backend) BatchSplitRegions(
 				retryRegions = append(retryRegions, region)
 				continue
 			}
-			if err = local.splitCli.ScatterRegion(ctx, region); err != nil {
```

Comment:
we have retry scatter after checking `hasRegion` previously, now it's done after `hasRegion` check of all regions. how this affect region skewness?

and i think change from `try scatter next region` to `try scatter same region` after some scatter failed due to store limit, does NOT reduce unnecessarily load on PD

Reply:
Why do we want to scatter other regions in the list If we hit a store limit on one scatter operation. In my view it should back off if one scatter operation fails

Reply:
If scatter operation fails It is adding the regions into retryRegions currently and keep doing scatter of other regions in the list.  It can cause lot of cancel scatter operation in PD.

Reply:
> Why do we want to scatter other regions in the list If we hit a store limit on one scatter operation.

i get it. but still not get why this affect region skewness.

as in `It doesn't take into number of scatter operations cancelled because of store limit.`, those cancelled operation will be retried and finally became be scheduled.

Reply:
There are two problems

In 6.5 , with parallel import and one region being split into 1000 regions , it hit the retry limit and failing to the scatter operations. It increase the retry limit only after scattering all the regions in the list and the retry limit is set to only 5.

PD scatter the regions on a node which has the least scatter operations scheduled. It doesn't take cancelled scatter regions into account. Lets say you there are 5 tikv node and you send 6 scatter operations than 1 and 5 can get scheduled to same node even though 2 to 4 scatter operations are cancelled.

I understand that master has some fixes like region-check-backoff-limit which is set to 1800 and it also ncreases the retry limit more conservatively which might fix some cases.

Reply:
> In 6.5

yes, it's a [known issue](https://github.com/pingcap/tidb/issues/43510), which might cause data skew on large data import, it's fixed on master/7.1.x, but not on to 6.5.x(as the pr introduces new parameters)

> However i think that for a complete solution , retry limit should be applied to individual regions rather than the whole list.

lgtm. now i know this pr only enhance this part, it's not related to region skewness

Reply:
can you review this PR